cranelift/codegen/src/opts/icmp.isle

```diff
@@ -0,0 +1,156 @@
+;; `icmp`-related rewrites
+
+;; Reflexivity/irreflexivity of `icmp` cond codes:
```

Comment:
It looks like these rules (for reflexive/irreflexive comparisons) are new; would you mind putting them in a separate PR, just to keep the logical changes separated?

Reply:
Looks like they aren't new but they are refactored. I agree with Chris that I'd rather discuss those changes in a separate PR, as I would rather not refactor them in quite this way. If you can move the rules verbatim this PR will be easier to review and get merged.